### ISSUE TITLE
Markdown links to markdown files will be broken if the file name does not contain space but is surrounded by <>

### ISSUE THREAD

**Issue by axmmisaka** [facebook/docusaurus#9614]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Per CommonMark, `[Haha](../hehe.mdx)` and `[Haha](<../hehe.mdx>)` should be the same (if the path to file contains space only the latter will work, nevertheless.)
In Docusaurus 3, if the file path contains space only the latter syntax will work; however, if the file path does not contain space, only the former syntax will work and the latter will result in broken link.

![image](https://github.com/facebook/docusaurus/assets/6500159/fb4fb4f5-a6a1-4542-b567-2d6d5fffb866)

### Steps to reproduce

1. Open docusaurus.new
2. In `docs/tutorial-basics/congratulations.md`,
```diff
- Have **5 more minutes**? Take a look at **[versioning](../tutorial-extras/manage-docs-versions.md)** and **[i18n](../tutorial-extras/translate-your-site.md)**.
+ Have **5 more minutes**? Take a look at **[versioning](<../tutorial-extras/manage-docs-versions.md>)** and **[i18n](<../tutorial-extras/translate-your-site.md>)**.
```
3. `yarn build`

### Expected behavior

As if nothing has changed

### Actual behavior

Broken links found

### Your environment

- Public source code:
- Public site URL:
- Docusaurus version used:
- Environment name and version (e.g. Chrome 89, Node.js 16.4): NodeJS 20
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by OzakIOne** [comment]
I've tried to create a repro [here](https://stackblitz.com/edit/github-mwgt1f?file=docs%2Fintro.md)

Indeed link surrounded by `<>` will be broken, however I could not get the link with space to work with `<>`

**Comment by axmmisaka** [comment]
> I've tried to create a repro [here](https://stackblitz.com/edit/github-mwgt1f?file=docs%2Fintro.md)
> 
> Indeed link surrounded by `<>` will be broken, however I could not get the link with space to work with `<>`

I checked this reproduction and these two links work on my end, which is the expected behaviour? The second link looks weird in the editor though.

```
[page](./page.md)
[page](./page.md)
[space](./page space.md)
[space](<./page space.md>) <-
[space](page space.md)
[space](<page space.md>) <-
[space](./pagespace.md)
```

**Comment by slorber** [comment]
hmmm 🤔 I thought we fixed this bug in https://github.com/facebook/docusaurus/issues/8867 but apparently it's back despite unit tests? Need to be investigated

**Comment by Josh-Cena** [comment]
I think the regex only matches links with both `<>` and spaces... Not sure exactly how

**Comment by axmmisaka** [comment]
> hmmm 🤔 I thought we fixed this bug in #8867 but apparently it's back despite unit tests? Need to be investigated

It was replaced by a plethora of big regexps which I found it powerful enough by looking at it while realising at the same time I am too stupid to comprehend it

https://github.com/facebook/docusaurus/blob/8dd1e13f2aceffcd82e2dcf1f409082fa99c2eff/packages/docusaurus-utils/src/markdownLinks.ts#L123-L137

**Comment by Josh-Cena** [comment]
@axmmisaka I've submitted a PR: https://github.com/facebook/docusaurus/pull/9617

The regex evolved over time. Now it's probably not a good idea to reason it from scratch😅
